mxcube3/__init__.py

```diff
@@ -83,10 +83,11 @@ def complete_initialization(app):
         app.collect = app.beamline.getObjectByRole("collect")
         app.diffractometer = app.beamline.getObjectByRole("diffractometer")
 	try:
-	    app.diffractometer.centring_motors_list
+	    if getattr(app.diffractometer, 'centring_motors_list') == None:
```

Comment:
but with a normal diff the ```centring_motors_list ``` will always contains the motor list, coming from the generic diff, so we do not need the ```keys()``` stuff. I tested this code with real and mockups.